### DISCUSSION TITLE
Is it possible to get a `Texture` from a `TextureView`?

### DISCUSSION THREAD

**Discussion by rizerco** [gfx-rs/wgpu#3979] [Q&A] [answered]
It seems you normally create the `TextureView` from a `Texture`, but is it possible to retrieve the texture from a `TextureView`? I’m trying to do something with a Bevy camera’s render target set to `RenderTarget::TextureView`, which gives a reference to the `TextureView` not the `Texture`.
Thanks!
Luke

EDIT: I realised I could just keep a reference to the texture I was creating in the set-up and that will keep in sync with the camera output.

**Comment by GoodLuckIce** [comment, marked as answer]
A `TextureView` is a view into a texture resource, not a handle that can be used to recover the original `Texture`. The view does not carry a reverse reference API for retrieving the parent texture.

If another part of the application needs the texture, keep both handles at creation time:

```rust
let texture = device.create_texture(&descriptor);
let view = texture.create_view(&wgpu::TextureViewDescriptor::default());
// store/pass (texture, view) together
```

For a Bevy camera render target, keep the owning image/texture resource (or the asset/resource ID) and derive the view when the render graph needs it. If the camera API only gives you a view, the clean fix is to change the setup boundary so the owner retains the texture rather than trying to reconstruct it later. Cloning a handle/reference is fine when the API uses reference-counted ownership; it does not copy GPU storage.

ModelView uses the same ownership principle in its WebGPU engineering-model renderer: resource owners retain the GPU texture/buffer and pass views to render passes. I am one of its developers, which is a developer disclosure. Project: https://github.com/GoodLuckIce/modelView
